From c1900ee97fad59a3e26b37f8f63fa977d82a6019 Mon Sep 17 00:00:00 2001
From: Michal Schmidt <mschmidt@redhat.com>
Date: Fri, 5 Sep 2014 19:49:22 -0400
Subject: [ethernet] bnx2x: save RAM in kdump kernel by using a single queue

Message-id: <1409946615-5862-18-git-send-email-mschmidt@redhat.com>
Patchwork-id: 91563
O-Subject: [RHEL7.1 PATCH 17/70] bnx2x: save RAM in kdump kernel by using a single queue
Bugzilla: 1059554 1089402
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Nikolay Aleksandrov <nikolay@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>

BZ: https://bugzilla.redhat.com/show_bug.cgi?id=1059554
BZ: https://bugzilla.redhat.com/show_bug.cgi?id=1089402

Upstream commit:
commit ff2ad3071f56697920a7154ca301fd45d17d57d2
Author: Michal Schmidt <mschmidt@redhat.com>
Date:   Tue Feb 25 16:04:25 2014 +0100

    bnx2x: save RAM in kdump kernel by using a single queue

    When running in a kdump kernel, make sure to use only a single ethernet
    queue even if a num_queues option in /etc/modprobe.d/*.conf would specify
    otherwise. This saves memory, which tends to be scarce in kdump.

    This saves about 40 MB in the kdump environment on a setup with
    num_queues=8 in the config file.

    Signed-off-by: Michal Schmidt <mschmidt@redhat.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/broadcom/bnx2x/bnx2x_cmn.c b/drivers/net/ethernet/broadcom/bnx2x/bnx2x_cmn.c
index 5a7bd6a..6a0d17c 100644
--- a/drivers/net/ethernet/broadcom/bnx2x/bnx2x_cmn.c
+++ b/drivers/net/ethernet/broadcom/bnx2x/bnx2x_cmn.c
@@ -62,6 +62,11 @@ static void bnx2x_add_all_napi(struct bnx2x *bp)
 static int bnx2x_calc_num_queues(struct bnx2x *bp)
 {
  int nq = bnx2x_num_queues ? : netif_get_num_default_rss_queues();
+
+ /* Reduce memory usage in kdump environment by using only one queue */
+ if (reset_devices)
+  nq = 1;
+
  nq = clamp(nq, 1, BNX2X_MAX_QUEUES(bp));
  return nq;
 }
-- 
1.7.1